File: SamCross/ExampleProjects/Math_Graphic/src/main/java/core/Math/Shapes/Circle.java
package  core.Math.Shapes;

/**
 * Created by Michal on 2018-04-07.
 */
import core.Comon.Bitmap;
import  core.Math.Shape;
import core.Math.ShapeType;

public class Circle extends Shape {
    PointXY center;
    int area = -1;
    int r;
    public Circle(PointXY center, int r){
        this.center = center;
        this.r = r;
        ShapeType = core.Math.ShapeType.CIRCLE;
    }
    @Override
    public Bitmap DrawMe() {
        area = 0;
        int array[][] = new int[1000][1000];
        for(int i=0; i<=2*r; i++){
            for(int j=0; j<=2*r; j++) {
                if((i - r) * (i - r) + (j - r) * (j - r) <= r*r){
                    array[origin().X+i][origin().Y+j] = 1;
                    area++;
                }
            }
        }
        return new Bitmap(array);
    }

    public PointXY origin(){
        return new PointXY(center.X-r, center.Y-r);
    }

    @Override
    public int Area(){
        if(area == -1) DrawMe();
        return area;
    }
}

File: stryker-net/ExampleProjects/Math_Graphic/src/Math_Graphic/core/math/shapes/Circle.cs
using Math_Graphic.core.common;

namespace Math_Graphic.core.math.shapes;

public class Circle : Shape
{
    private readonly PointXy _center;
    private int _area = -1;
    private readonly int _r;

    public Circle(PointXy center, int r)
    {
        _center = center;
        _r = r;
        ShapeType = ShapeType.Circle;
    }

    public override Bitmap DrawMe()
    {
        _area = 0;
        var array = new int[1000, 1000];
        for (var i = 0; i <= 2 * _r; i++)
        {
            for (var j = 0; j <= 2 * _r; j++)
            {
                if ((i - _r) * (i - _r) + (j - _r) * (j - _r) > _r * _r) continue;
                array[Origin().X + i, Origin().Y + j] = 1;
                _area++;
            }
        }

        return new Bitmap(array);
    }

    public PointXy Origin()
    {
        return new PointXy(_center.X - _r, _center.Y - _r);
    }

    public override int Area()
    {
        if (_area == -1) DrawMe();
        return _area;
    }
}